insights/tests/client/test_client.py

```diff
@@ -329,10 +330,50 @@ def test_upload_500_retry(_, upload_archive):
 
         upload_archive.assert_called()
         assert upload_archive.call_count == retries
+        logger.error.assert_any_call("Upload attempt %d of %d failed! Reason: %s", 1, config.retries, ANY)
+        logger.error.assert_called_with("All attempts to upload have failed!")
     finally:
         sys.argv = tmp
 
 
+@patch('insights.client.client.constants.sleep_time', 0)
+@patch('insights.client.client.InsightsConnection.upload_archive')
+@patch('insights.client.os.path.exists', return_value=True)
```

Comment:
I'm a bit afraid to mock the path.exists method, as it is use in a lot of places of this code. I have experience some unwanted failures because of that. If this is used to check if a file exists I would suggest to use @contexmanager to create a temporary file to check against.

Reply:
Good idea.  I removed the os.path.exists mock and used a contextmanager for managing the temporary archive file.  In my defence though, the surrounding methods mock the os.path.exists method :)

Reply:
Actually, the tests didn't like the contextmanager for the temporary archive because the file didn't exist.  Instead I've moved the os.path.exists mock to only surround the upload() call, thus limiting its scope, instead of using it for the whole test.  Anyway, as I mentioned before, I was just copying what the other tests had done.

Reply:
oh, you can created as this: https://github.com/RedHatInsights/insights-core/blob/master/insights/tests/client/test_client.py#L70, I'm using it to create the registration test up in this test